snapshot.go

```diff
@@ -0,0 +1,254 @@
+package apidApigeeSync
+
+import (
+	"github.com/30x/apid-core"
+	"net/http"
+	"encoding/json"
+	"os"
+	"github.com/30x/apid-core/data"
+	"github.com/apigee-labs/transicator/common"
+
+	"io"
+	"time"
+	"net/url"
+	"path"
+)
+
+// retrieve boot information: apid_config and apid_config_scope
+func downloadBootSnapshot(quitPolling chan bool) {
+	log.Debug("download Snapshot for boot data")
+
+	scopes := []string{apidInfo.ClusterID}
+	snapshot := &common.Snapshot{}
+	downloadSnapshot(scopes, snapshot, quitPolling)
+	// note that for boot snapshot case, we don't need to inform plugins as they'll get the data snapshot
+	processSnapshot(snapshot)
+}
+
+// use the scope IDs from the boot snapshot to get all the data associated with the scopes
+func downloadDataSnapshot(quitPolling chan bool) {
+	log.Debug("download Snapshot for data scopes")
+
+	var scopes = findScopesForId(apidInfo.ClusterID)
+	scopes = append(scopes, apidInfo.ClusterID)
+	snapshot := &common.Snapshot{}
+	downloadSnapshot(scopes, snapshot, quitPolling)
+
+	knownTables = extractTablesFromSnapshot(snapshot)
+
+	db, err := dataService.DBVersion(snapshot.SnapshotInfo)
+	if err != nil {
+		log.Panicf("Database inaccessible: %v", err)
+	}
+	persistKnownTablesToDB(knownTables, db)
+
+	done := make(chan bool)
+	log.Info("Emitting Snapshot to plugins")
+	events.EmitWithCallback(ApigeeSyncEventSelector, snapshot, func(event apid.Event) {
```

Comment:
Try to pull the new apid-core, and use the new blocking Emit() with returned channel for this:

```
select {
 	case <-time.After(pluginTimeout):
 		log.Panic("Timeout. Plugins failed to respond to snapshot.")
 	case <-events.Emit(ApigeeSyncEventSelector, snapshot):
}
```
No need for the `done` chan